ir/memory.cpp

```diff
@@ -673,9 +693,15 @@ expr Pointer::is_writable() const {
   return is_local() || non_local;
 }
 
+expr Pointer::is_nocapture() const {
+  if (bits_for_ptrattrs == 0)
+    return false;
+  return get_attrs() == 1;
```

Comment:
this is not correct once we have more attributes.  I would do an extract here directly and remove get_attrs()